connect_ext_test.go

```diff
@@ -2442,97 +2483,6 @@ func TestClientDisconnect(t *testing.T) {
 	})
 }
 
-func TestTrailersOnlyErrors(t *testing.T) {
```

Comment:
These tests were for the old classification and no longer apply.

For example, they'd check that if we classify a response as trailers-only (due to presence of grpc-status key), then the client complains (in the form of an RPC error with "internal" code) if there is a non-empty body or non-empty HTTP trailers.

But since we now define trailers-only to be a request without body or HTTP trailers, they are moot. In these situations (where a response has a grpc-status key in headers and a non-empty body) we now just ignore the grpc-status key in headers (which is also what grpc-go does).